src/katsdpcontroller/product_config.py

```diff
@@ -1227,6 +1227,56 @@ def from_config(
         )
 
 
+class GpucbfTiedArrayChannelisedVoltageStream(TiedArrayChannelisedVoltageStreamBase):
+    """Real tied-array-channelised-voltage stream (GPU beamformer)."""
+
+    stream_type: ClassVar[str] = "gpucbf.tied_array_channelised_voltage"
+    _valid_src_types: ClassVar[_ValidTypes] = {"gpucbf.antenna_channelised_voltage"}
+
+    def __init__(
+        self,
+        name: str,
+        src_streams: Sequence[Stream],
+        *,
+        src_pol: int,
+        command_line_extra: Iterable[str] = (),
+    ) -> None:
+        acv = src_streams[0]
+        assert isinstance(acv, GpucbfAntennaChannelisedVoltageStream)
+        super().__init__(
+            name,
+            src_streams,
+            n_endpoints=acv.n_substreams,
+            n_chans_per_substream=acv.n_chans_per_substream,
+            spectra_per_heap=acv.n_spectra_per_heap,
+            bits_per_sample=8,
+        )
+        self.src_pol = src_pol
+        self.command_line_extra = list(command_line_extra)
+
+    if TYPE_CHECKING:  # pragma: nocover
+        # Refine the return type for mypy
+        @property
+        def antenna_channelised_voltage(self) -> AntennaChannelisedVoltageStreamBase:
```

Comment:
I think this should be
```suggestion
        def antenna_channelised_voltage(self) -> GpucbfAntennaChannelisedVoltageStream:
```
Otherwise it's not adding any value